### ISSUE TITLE
[BUG] TelnetOOB encode_gmcp() cmdname formatting incompatible with Mudlet's Client.GUI command

### ISSUE THREAD

**Issue by eartsar** [evennia/evennia#2300]
#### Describe the bug
The Mudlet client supports customizing its user interface through GMCP commands. One in particular, `Client.GUI`, gives a directive for the client to download an `mpackage` file used for customizing the user interface. As of the latest version of the client, interpretation of this command seems to be case-sensitive.

`TelnetOOB::encode_gmcp()` has extensive documentation on the formatting procedures put in place for adjusting the command name, presumably to enforce a standard. However, use of `str.capitalize()` makes it impossible to send `Client.GUI` verbatim, instead yielding `Client.Gui`. Perhaps this is not on Evennia to fix, but introducing a way to work around the issue would be idea, by either allowing a flag that turns off the auto-formatting of `cmdname`, or by adding to the `EVENNIA_TO_GMCP` mapping.

#### To Reproduce
1. Send the necessary OOB command (however you'd like). For example: `obj.msg(Client_GUI=([], {'version': '1.0', 'url': '127.0.0.1:8080'}))` Feel free to change the URL to any mpackage.
2. Connect to the server via Mudlet. Open up the debugger to see intepretation of the GMCP command.
3. Trigger the OOB message, and nothing happens.

#### Expected behavior
Mudlet should download the mpackage from the given URL, and apply it in its installed packages.

#### Environment, Evennia version, OS etc
```
    Evennia 0.9.5
    OS: posix
    Python: 3.9.1
    Twisted: 20.3.0
    Django: 2.2.17
    ```

**Comment by Griatch** [comment]
AFAIK, capitalized is the most commonly used form for GMCP message names, so arguably the client should really be requesting Gui and not GUI to go with existing standards. But like everything else in MUD world the specs are a bit up to interpretation and there is no central authority to enforce standards. So the easiest way to handle the GUI case may be to have Evennia analyze the name and not run ´.capitalize()ˋ on it when the name is already capitalized.

**Comment by eartsar** [comment]
> AFAIK, capitalized is the most commonly used form for GMCP message names, so arguably the client should really be requesting Gui and not GUI to go with existing standards. But like everything else in MUD world the specs are a bit up to interpretation and there is no central authority to enforce standards. So the easiest way to handle the GUI case may be to have Evennia analyze the name and not run ´.capitalize()ˋ on it when the name is already capitalized.

I have a monkey-patch work-around, so this fix isn't crucial to me for this repository to have. Indeed, GMCP documentation indicates this should be case-insensitive (at least, I interpret it this way). I've opened up an issue with the Mudlet repo (https://github.com/Mudlet/Mudlet/issues/4566).

**Comment by vadi2** [comment]
We'll fix it in the next update of Mudlet :+1: I recommend still sending it as `Client.GUI` for older installations out there.

**Comment by Griatch** [comment]
Implemented in `develop` branch (for Evennia 1.0).
